### PR TITLE
Add operations with data streams

### PR DESCRIPTION
**Pull request by zethuman**
### Description
Implemented Data Stream API: 
 * Create a data stream
 * Delete a data stream
 * Get all data streams
 * Get specific data stream
 * Get Stats of specific data stream  

Planned to add:
* Ingest data into the data stream
* Searching a data stream
* Rollover a data stream

### Issues Resolved
Closes [#152]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearchapi/api.indices.create_datastream.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+		params map[string]string
+	)
+
+	method = "PUT"
+
+	path.Grow(1 + len("_data_stream") + 1 + len(r.Name))
+	path.WriteString("/")
+	path.WriteString("_data_stream")
+	path.WriteString("/")
+	path.WriteString(r.Name)
```
Comment (dblock) [inline opensearchapi/api.indices.create_datastream.go:86]: This path is hard-coded except for `r.Name`, can we collapse into 2 `WriteString`?

**Review by dblock — COMMENTED** [review]
Good stuff. Needs tests, CHANGELOG, USER_GUIDE.

File: opensearchapi/api.indices.create_datastream.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+			f(&r)
+		}
+		return r.Do(r.ctx, t)
+	}
+}
+
+// ----- API Definition -------------------------------------------------------
+
+// IndicesCreateDataStream creates a data stream.
+//
```
[the remark this answers is not in the document]
Comment (zethuman) [reply opensearchapi/api.indices.create_datastream.go]: removed

File: opensearchapi/api.indices.create_datastream.go (hunk 1 above)
Comment (zethuman) [reply opensearchapi/api.indices.create_datastream.go:59]: fixed

**Comment by dblock** [comment]
Looks like some failures in tests, but otherwise looks good!

**Comment by zethuman** [comment]
Yes, my bad, did not look at the instructions on the Makefile. It seems that I wrote not unit tests, so thinking about adding integration tags, since I need an opensource instance for the test. What do you think?

**Pull request merged**